apps/src/aiTutor/redux/aiTutorRedux.ts

```diff
@@ -59,41 +66,26 @@ export const askAITutor = createAsyncThunk(
       scriptId: state.aiTutor.scriptId,
     };
 
-    const tutorType = chatContext.tutorType;
-    const generalChat = tutorType === TutorTypes.GENERAL_CHAT;
-    const compilation = tutorType === TutorTypes.COMPILATION;
-    const validation = tutorType === TutorTypes.VALIDATION;
-
-    let systemPrompt;
-    if (validation) {
-      systemPrompt = validationSystemPrompt;
-    } else if (compilation) {
-      systemPrompt = compilationSystemPrompt;
-    } else {
-      systemPrompt = generalChatSystemPrompt;
-    }
-
     const storedMessages = state.aiTutor.chatMessages;
 
     const newMessageId = storedMessages[storedMessages.length - 1].id + 1;
 
-    if (generalChat) {
-      const newMessage: ChatCompletionMessage = {
-        id: newMessageId,
-        role: Role.USER,
-        status: Status.UNKNOWN,
-        chatMessageText: chatContext.studentInput,
-      };
-      thunkAPI.dispatch(addChatMessage(newMessage));
-    }
+    const newMessage: ChatCompletionMessage = {
```

Comment:
Now, every scenario (general chat/compilation/validation) adds its "question" (the inputted question/"Why doesn't my code compile?"/"Why aren't my tests passing?") to state.